hazelcast-sql/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -0,0 +1,269 @@
+}
+{
+    type = QueryDataType() { span = span(); }
+    {
+        return new SqlDataType(type, span.end(this));
+    }
+}
+
+QueryDataType QueryDataType() :
+{
+    QueryDataType type;
+}
+{
+    (
+        type = NumericType()
+    |
+        type = CharacterType()
+    |
+        type = DateTimeType()
+    )
+    {
+        return type;
+    }
+}
+
+QueryDataType NumericType() :
+{
+    QueryDataType type;
+    int precision = -1;
+    int scale = -1;
+}
+{
+    (
+        <BOOLEAN> { type = QueryDataType.BOOLEAN; }
+    |
+        <TINYINT> { type = QueryDataType.TINYINT; }
+    |
+        <SMALLINT> { type = QueryDataType.SMALLINT; }
+    |
+        <INT> { type = QueryDataType.INT; }
+    |
+        <BIGINT> { type = QueryDataType.BIGINT; }
+    |
+        <REAL> { type = QueryDataType.REAL; }
+    |
+        <DOUBLE> { type = QueryDataType.DOUBLE; }
+    |
+        <DECIMAL> { type = QueryDataType.DECIMAL; }
+    )
+    {
+        return type;
+    }
+}
+
+QueryDataType CharacterType() :
+{
+    QueryDataType type;
+}
+{
+    (
+        <VARCHAR> { type = QueryDataType.VARCHAR; }
+    )
+    {
+        return type;
+    }
+}
+
+QueryDataType DateTimeType() :
+{
+    QueryDataType type;
+    SqlIdentifier variant = null;
+}
+{
+    (
+        <TIME> { type = QueryDataType.TIME; }
+    |
+        <DATE> { type = QueryDataType.DATE; }
+    |
+        <TIMESTAMP>
+        (
+            <WITH> <TIME> <ZONE> { type = QueryDataType.TIMESTAMP_WITH_TZ_OFFSET_DATE_TIME; }
+        |
+            { type = QueryDataType.TIMESTAMP; }
+        )
+    )
+    {
+        return type;
+    }
+}
+
+/**
+* Parses OPTIONS.
+*/
+SqlNodeList SqlOptions():
+{
+    Span span;
+
+    Map<String, SqlNode> sqlOptions = new LinkedHashMap<String, SqlNode>();
+    SqlOption sqlOption;
+}
+{
+    <LPAREN> { span = span(); }
+    [
+        sqlOption = SqlOption()
+        {
+            sqlOptions.put(sqlOption.key(), sqlOption);
+        }
+        (
+            <COMMA> sqlOption = SqlOption()
+            {
+                if (sqlOptions.putIfAbsent(sqlOption.key(), sqlOption) != null) {
+                    throw SqlUtil.newContextException(getPos(),
+                        ParserResource.RESOURCE.duplicateOption(sqlOption.key()));
+                }
+            }
+        )*
+    ]
+    <RPAREN>
+    {
+        return new SqlNodeList(sqlOptions.values(), span.end(this));
+    }
+}
+
+SqlOption SqlOption() :
+{
+    Span span;
+
+    SqlIdentifier key;
+    SqlNode value;
+}
+{
+    key = SimpleIdentifier() { span = span(); }
+    value = StringLiteral()
+    {
+        return new SqlOption(key, value, span.end(this));
+    }
+}
+
+/**
+* Parses DROP EXTERNAL TABLE statement.
+*/
+SqlDrop SqlDropExternalTable(Span span, boolean replace) :
+{
+    SqlParserPos pos = span.pos();
+
+    SqlIdentifier name;
+    boolean ifExists = false;
+}
+{
+    <EXTERNAL> <TABLE>
+    [
+        <IF> <EXISTS> { ifExists = true; }
+    ]
+    name = SimpleIdentifier()
```

Comment:
Change it to `CompundIdentifier` and validate the schema, if present, is equal to `public`? https://github.com/hazelcast/hazelcast/pull/17266#discussion_r460783679